src/sentry/api/endpoints/team_all_unresolved_issues.py

```diff
@@ -18,24 +18,31 @@
 CLOSED_STATUSES = RESOLVED_STATUSES + (GroupHistoryStatus.IGNORED,)
 
 
-def calculate_unresolved_counts(team, project_list, start, end):
+def calculate_unresolved_counts(team, project_list, start, end, environment):
     # Get the current number of unresolved issues. We can use this value for the the most recent bucket.
+    group_enviornment_filter = (
+        {"groupenvironment__environment_id": environment.id} if environment else {}
+    )
     project_current_unresolved = {
         r["project"]: r["total"]
         for r in (
             Group.objects.filter_to_team(team)
-            .filter(status=GroupStatus.UNRESOLVED)
+            .filter(status=GroupStatus.UNRESOLVED, **group_enviornment_filter)
             .values("project")
             .annotate(total=Count("id"))
         )
     }
 
+    group_history_enviornment_filter = (
+        {"group__groupenvironment__environment_id": environment.id} if environment else {}
+    )
     prev_status_sub_qs = Coalesce(
         Subquery(
             GroupHistory.objects.filter(
                 group_id=OuterRef("group_id"),
                 date_added__lt=OuterRef("date_added"),
                 status__in=OPEN_STATUSES + CLOSED_STATUSES,
+                **group_history_enviornment_filter,
```

Comment:
We don't need this for the subquery - Any groups here will already be filtered by the top level query, so we know that they belong to the environment.